xarray/core/alignment.py

```diff
@@ -358,7 +366,15 @@ def var_indexers(var, indexers):
             args = (var.attrs, var.encoding)
         else:
             args = ()
-        reindexed[dim] = IndexVariable((dim,), indexers[dim], *args)
+
+        idx_var = IndexVariable((dim,), indexers[dim], *args)
+
+        # GH1434
+        # ensures that dtype of numpy structured arrays is preserved
+        if len(args) and is_composite_dtype(var.dtype) \
+                and idx_var.dtype != var.dtype:
+            idx_var.data = idx_var.data.astype(var.dtype)
```

Comment:
OK, this does indeed make sense. To isolate he issue from your example, let's see what we get from aligning instead:
```
>>> da1_aligned, da2_aligned = xarray.align(da1, da2, join='outer')
>>> da1
<xarray.DataArray (measurement: 10, participant: 3)>
array([[ 50.,  51.,  52.],
       [ 53.,  54.,  55.],
       [ 56.,  57.,  58.],
       [ 59.,  60.,  61.],
       [ 62.,  63.,  64.],
       [ 65.,  66.,  67.],
       [ 68.,  69.,  70.],
       [ 71.,  72.,  73.],
       [ 74.,  75.,  76.],
       [ 77.,  78.,  79.]])
Coordinates:
  * measurement  (measurement) int64 0 1 2 3 4 5 6 7 8 9
  * participant  (participant) [('name', 'S256'), ('height', '<i8')] ('A', 180) ...

>>> da1_aligned
<xarray.DataArray (measurement: 10, participant: 6)>
array([[ 50.,  51.,  52.,  nan,  nan,  nan],
       [ 53.,  54.,  55.,  nan,  nan,  nan],
       [ 56.,  57.,  58.,  nan,  nan,  nan],
       [ 59.,  60.,  61.,  nan,  nan,  nan],
       [ 62.,  63.,  64.,  nan,  nan,  nan],
       [ 65.,  66.,  67.,  nan,  nan,  nan],
       [ 68.,  69.,  70.,  nan,  nan,  nan],
       [ 71.,  72.,  73.,  nan,  nan,  nan],
       [ 74.,  75.,  76.,  nan,  nan,  nan],
       [ 77.,  78.,  79.,  nan,  nan,  nan]])
Coordinates:
  * participant  (participant) object ('A', 180) ('B', 150) ('C', 200) ...
  * measurement  (measurement) int64 0 1 2 3 4 5 6 7 8 9
```
Notice that the dtype of `participant` is now `object.